docs/user/general/datatypes.rst

```diff
@@ -354,18 +353,94 @@ A string can also represent and be converted to date and time types (except to i
     +------------------------------------------------------------+-------------------------------------+----------------------------------+
 
+Please, see `more examples here <../dql/expressions.rst#toc-entry-15>`_.
+
+Date formats
+------------
+
+SQL plugin supports all named formats for OpenSearch ``date`` data type, custom formats and their combination. Please, refer to `OpenSearch docs <https://opensearch.org/docs/latest/field-types/supported-field-types/date/>`_ for format description.
+Plugin detects which type of data is stored in ``date`` field according to formats given and returns results in the corresponding SQL types.
+Given an index with the following mapping.
+
+.. code-block:: json
+
+    {
+        "mappings" : {
+            "properties" : {
+                "date1" : {
+                    "type" : "date",
+                    "format": "yyyy-MM-dd"
+                },
+                "date2" : {
+                    "type" : "date",
+                    "format": "date_time_no_millis"
+                },
+                "date3" : {
+                    "type" : "date",
+                    "format": "hour_minute_second"
+                },
+                "date4" : {
+                    "type" : "date"
+                },
+                "date5" : {
+                    "type" : "date",
+                    "format": "yyyy-MM-dd || time"
+                }
+            }
+        }
+    }
+
+Querying such index will provide a response with ``schema`` block as shown below.
+
+.. code-block:: json
+
+    {
+        "query" : "SELECT * from date_formats LIMIT 0;"
+    }
+
+.. code-block:: json
+
+    {
+        "schema": [
+            {
+                "name": "date5",
+                "type": "timestamp"
+            },
+            {
+                "name": "date4",
+                "type": "timestamp"
+            },
+            {
+                "name": "date3",
+                "type": "time"
+            },
+            {
+                "name": "date2",
+                "type": "timestamp"
+            },
+            {
+                "name": "date1",
+                "type": "date"
+            },
+        ],
+        "datarows": [],
+        "total": 0,
+        "size": 0,
+        "status": 200
+    }
+
 String Data Types
 =================
 
-A string is a sequence of characters enclosed in either single or double quotes. For example, both 'text' and "text" will be treated as string literal. To use quote characters in a string literal, you can use two quotes of the same type as the enclosing quotes::
+A string is a sequence of characters enclosed in either single or double quotes. For example, both 'text' and "text" will be treated as string literal. To use quote characters in a string literal, you can use two quotes of the same type as the enclosing quotes or a backslash symbol (``\``)::
```

Comment:
<3

We might want to include an example of `'I''m'` also converting to `I'm` (if the text is quoted in the same type of quote (single or double) as is used twice in a row, it acts as an escape as well).